__tests__/api.test.js

```diff
@@ -0,0 +1,291 @@
+/* eslint-disable no-underscore-dangle */
+require('dotenv').config();
+const supertest = require('supertest');
+const mongoose = require('mongoose');
+const app = require('../server/api.js');
+const { InstructorsModel, OfferedBysModel, TestimonialsModel } = require('../db/models.js');
+const { instructorsInsert, offeredBysInsert, testimonialsInsert } = require('../db/methods.js');
+const instructorsData = require('../db/data/instructors.json');
+const offeredBysData = require('../db/data/offeredBys.json');
+const testimonialsData = require('../db/data/testimonials.json');
+
+const request = supertest(app);
+
+beforeAll(async (done) => {
+  mongoose.connect(`mongodb://${process.env.DB_HOSTNAME}:${process.env.DB_PORT}/test`, { useNewUrlParser: true, useUnifiedTopology: true });
+  await InstructorsModel.deleteMany({});
+  await OfferedBysModel.deleteMany({});
+  await TestimonialsModel.deleteMany({});
+  await instructorsInsert(instructorsData);
+  await offeredBysInsert(offeredBysData);
+  await testimonialsInsert(testimonialsData);
+  done();
+});
+
+afterAll(async (done) => {
+  await mongoose.connection.close();
+  done();
+});
+
+describe('Test CRUD requests for instructors', () => {
+  test('should create one new document in the instructors database', async (done) => {
```

Comment:
We discussed this during stand-ups, but it might be beneficial to add tests for errors / failures. I like how you broke up your tests into sections for instructors, offeredBy, and testimonials.

Reply:
Thank you for noticing. Currently it all responds with a 200. I plan to fix that handling from within the DB methods.